src/componenets/header.js

```diff
@@ -0,0 +1,27 @@
+// TEST
+
+class foodAppHeader extends HTMLElement {
+  constructor() {
+    super();
+
+  }
+
+  connectedCallback() {
+    this.innerHTML = `
+    <header class="nav-header">
+      <nav class="nav-bar">
+        <ul class="nav-list">
+          <li class="nav-item"><a href="#">Home</a></li>
+          <li class="nav-item"><a href="#">Home</a></li>
+          <li class="nav-item"><a href="#">Home</a></li>
+          <li class="nav-item"><a href="#">Home</a></li>
+          <li class="nav-item"><a href="#">Home</a></li>
+          <li class="nav-item"><a href="#">Home</a></li>
+        </ul>
+      </nav>
+    </header>
+    `;
+  }
+};
+
+customElements.define('component-header', foodAppHeader); // first param is html tag
```

Comment:
this use of web components is spot on, you can try to pass array of available links and them use them to build this unordered list